### PR TITLE
Fix/w3c compatibility issues

### PR DESCRIPTION
**Pull request by basdijkstra**
This PR fixes W3C compatibility issues that caused a number of problems when using, for example, `By.Id` locators and Selenium commands that invoke the `executeScript` WebDriver W3C command.

### CODE REVIEW

File: TestProject.OpenSDK/Drivers/BaseDriver.cs [hunk 1]
```diff
@@ -83,16 +84,22 @@ static BaseDriver()
...
-            sessionIdBase.SetValue(this, this.sessionId);
+            FieldInfo sessionIdField = typeof(RemoteWebDriver).GetField("sessionId", BindingFlags.Instance | BindingFlags.NonPublic);
+            sessionIdField.SetValue(this, new SessionId(this.sessionId));
 
             // Create a new command executor for this driver session and set disable reporting flag
             this.commandExecutor = new CustomHttpCommandExecutor(AgentClient.GetInstance().AgentSession.RemoteAddress, disableReports);
 
+            // If the driver returned by the Agent is in W3C mode, we need to update the command info repository
+            // associated with the base RemoteWebDriver to the W3C command info repository (default is OSS).
+            if (AgentClient.GetInstance().AgentSession.Dialect.Equals("W3C"))
```
Comment (eldarkra) [inline TestProject.OpenSDK/Drivers/BaseDriver.cs:95]: I see that `AgentClient.GetInstance().AgentSession.Dialect.Equals("W3C")` is being used on several occasions. I think that an extension method (something like `AgentClient.IsW3CDialect()`) can be created.

File: TestProject.OpenSDK/Drivers/BaseDriver.cs [hunk 2]
```diff
@@ -173,21 +180,20 @@ protected override Response Execute(string driverCommandToExecute, Dictionary<st
...
                 resp.SessionId = this.sessionId;
                 resp.Value = new Dictionary<string, object>();
+
                 return resp;
             }
 
             // The Agent does not understand the default way Selenium sends the driver command parameters for SendKeys
             // This means we'll need to patch them so these commands can be executed.
-            if (driverCommandToExecute.ShouldBePatched())
+            if (!AgentClient.GetInstance().AgentSession.Dialect.Equals("W3C") && driverCommandToExecute.ShouldBePatched())
```
Comment (eldarkra) [inline TestProject.OpenSDK/Drivers/BaseDriver.cs:189]: Use the extension method

File: TestProject.OpenSDK/Internal/Helpers/CommandExecutors/CustomHttpCommandExecutor.cs [hunk 3]
```diff
@@ -50,6 +52,14 @@ public CustomHttpCommandExecutor(Uri addressOfRemoteServer, bool disableReports)
         public CustomHttpCommandExecutor(Uri addressOfRemoteServer, bool disableReports, TimeSpan remoteConnectionTimeout)
             : base(addressOfRemoteServer, remoteConnectionTimeout)
         {
+            // If the driver returned by the Agent is in W3C mode, we need to update the command info repository
+            // associated with the base HttpCommandExecutor to the W3C command info repository (default is OSS).
+            if (AgentClient.GetInstance().AgentSession.Dialect.Equals("W3C"))
```
Comment (eldarkra) [inline TestProject.OpenSDK/Internal/Helpers/CommandExecutors/CustomHttpCommandExecutor.cs:57]: Use the extension method

**Review by eldarkra — CHANGES_REQUESTED** [review]

**Comment by basdijkstra** [comment]
@eldarkra added and applied the requested extension method, please re-review.

**Review by eldarkra — APPROVED** [review]

**Comment by eldarkra** [comment]
@basdijkstra looks OK but now the brach is out of date once again.

**Comment by basdijkstra** [comment]
> @basdijkstra looks OK but now the brach is out of date once again.

No problem, I'll rebase and merge it myself. Thanks.

**Pull request merged**